src/test/java/LinkedIteratorTest.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class LinkedIteratorTest {
+
+    @Test
+    void hasNext_emptyLinked_returnsFalse() {
+        Node<Integer> head = null;
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        assertFalse(it.hasNext());
+    }
+
+    @Test
+    void hasNext_singleElementLinked_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        Iterator<Integer> it = new LinkedIterator<>(head);
+        assertTrue(it.hasNext());
+    }
+
+    @Test
+    void hasNext_multiElementLinked_returnsTrue() {
+        Node<Integer> head = new Node<>(10);
+        head.setNext(new Node<>(11));
+        head.getNext().setNext(new Node<>(12));
+        head.getNext().getNext().setNext(new Node<>(13));
+        head.getNext().getNext().getNext().setNext(new Node<>(14));
```

Comment:
You can save a lot of repeated code here with some private methods. Eg, I'd probably define:
```java
private Node<T> makeList(T t1, T... ts) {
    Node<T> head = new Node<>(t1);
    Node<T> current = head;
    for (int i = 0; i < ts.length; ++i) {
        Node<T> node = new Node<>(ts[i]);
        current.setNext(node);
        current = node;
    }
    return head;
}
```
Then you can write `Node<Integer> head = makeList(10, 11, 12, 13, 14)` in tests.